### ISSUE TITLE
includet and include don't actually do the same thing, and they should.

### ISSUE THREAD

**Issue by andyDoucette** [timholy/Revise.jl#682]
# Issue 
include() evaluates the contents of the input source in the global scope of the module you're in.

includet() evaluates the contents of the input source in the Main module only.

This is a major source of issues, as users like me will expect includet() to be a "tracking" version of include, otherwise the same.

I just spent 3 hours debugging an issue where if I included code using Genie (which used a different Module than Main), the included methods were nowhere to be found, but if I included the code directly from the REPL it worked fine.  It was challenging to figure out what's going on.  

# Request
Can we change the behavior of includet to use the module it's in instead of Main all the time?  Should be as easy as using @__MODULE__() I hope?

**Comment by pfitzseb** [comment]
I don't think there's a way for `includet` to stay a function _and_ get its caller's scope. You can of course already use `includet(@__MODULE__, "myfile.jl")`.

**Comment by andyDoucette** [comment]
If I can create a way, and make a pull request, would you merge it?

**Comment by timholy** [comment]
Impossible to say in the absence of the PR. But we merge PRs all the time here and in other packages.

Good PRs include a test and a clear motivation, and in this case the fact that you can use `includet(@__MODULE__, "myfile.jl")` seems like it undermines the need?

**Comment by Sixzero** [comment]
Also LanguageServer doesn't track functions included in `includet`. Which I think affects every developer using `includet` but most of us just still keeps going even if such functionalities break and use find in the project! 

As for the simplest MVP/fix is to change every includet to include and LanguageServer linkings will work, so LS navigations will work again. 

This is a serious issue, I don't know if the two are connected, or I should just open a separate issue.

**Comment by timholy** [comment]
That's funny because VSCode bundles Revise. So it knows about `includet`, and should be able to track things included that way.

**Comment by MasonProtter** [comment]
We could just make an `@includet` macro?

**Comment by pfitzseb** [comment]
Sure, but I still don't quite see the motivation for this.

**Comment by MasonProtter** [comment]
I don't think this should be closed. `includet` not evaluating into the module its located in is still quite the footgun.

**Comment by timholy** [comment]
Since there's no real technical reason against it, `@includet` is coming. But let me be clear: `includet` is intended for short-term, let-me-get-this-going development. The slippery slope of data-vs-code, the fact that `include` works as it does because it gets defined separately for each module, mean that `includet` is *already* its own footgun. I'm more than half-tempted to delete it rather than make it easier to use.
